File: Arrays/Leader/leaderOfArray.cpp
#include<iostream>
using namespace std;

void leader(int arr[], int n){

    int leader = arr[n-1];
    int count = 0;
    int *temp = new int(n);
    temp[0] = count;
    for(int i = n-2; i>=0; i--){
        if(leader < arr[i]){
            count++;
            leader = arr[i]; 
            temp[count] = leader;
        }
    }
    int tempSize = sizeof(temp)/sizeof(temp[0]);
    for(int i =0;i<tempSize; i++){
        cout<<temp[i]<<" ";
    }
}

int main()
{
    int arr[] = {2,10,4,10,6,5,2};
    int n = sizeof(arr)/sizeof(arr[0]);
    leader(arr,n);
    return 0;
}

File: Arrays/Leader/leaderOfArray.java
package Arrays.Leader;

import java.util.ArrayList;
import java.util.Collections;

public class leaderOfArray {

    public static void leader(Integer arr[], Integer n){
        Integer leader = arr[n-1];
        ArrayList<Integer> temp = new ArrayList<Integer>();
        temp.add(leader);
        for(int i = n-2; i>=0; i--){
            if(leader < arr[i]){
                leader = arr[i];
                temp.add(leader);
            }
        }
        Collections.reverse(temp);
        for(Integer x: temp){
            System.out.println(x);
        }
    }

    public static void main(String args[]){
        Integer arr[] = {2,10,4,3,10,6,5,2};
        leader(arr, arr.length);
    }
}
